test-data/unit/check-classes.test

```diff
@@ -2051,6 +2051,16 @@ class X: pass
 foo(X)
 [out]
 
+[case testTypeUsingTypeCTypeAnyMember]
+from typing import Type, Any
+def foo(arg: Type[Any]):
+    x = arg.member_name
```

Comment:
I can add a test with an assignment into a member of `Type[Any]` (from what I see, works with no changes). Is that what you're asking for?

I tried doing the fallback to `type` and it seems to work with small examples outside the test suite, but I'm having trouble writing the tests for that. Apparently `reveal_type(type.mro)` does different thing outside and inside the test suite (inside the test suite I get `Any` instead of `def () -> builtins.list[builtins.type]`) Should I do something special to get the right behaviour inside the test suite?